src/jit/valuenum.cpp

```diff
@@ -425,61 +256,513 @@ TFp FpRem(TFp dividend, TFp divisor)
     return (TFp)fmod((double)dividend, (double)divisor);
 }
 
-// Specialize for double for floating operations, that doesn't involve unsigned.
-template <>
-double ValueNumStore::EvalOp<double>(VNFunc vnf, double v0, double v1, ValueNum* pExcSet)
+VNFunc GetVNFuncForOper(genTreeOps oper, bool isUnsigned)
 {
-    genTreeOps oper = genTreeOps(vnf);
-    // Here we handle those that are the same for floating-point types.
+    if (!isUnsigned || (oper == GT_EQ) || (oper == GT_NE))
+    {
+        return VNFunc(oper);
+    }
     switch (oper)
     {
+        case GT_LT:
+            return VNF_LT_UN;
+        case GT_LE:
+            return VNF_LE_UN;
+        case GT_GE:
+            return VNF_GE_UN;
+        case GT_GT:
+            return VNF_GT_UN;
         case GT_ADD:
-            return FpAdd<double, DoubleTraits>(v0, v1);
+            return VNF_ADD_UN;
         case GT_SUB:
-            return FpSub<double, DoubleTraits>(v0, v1);
+            return VNF_SUB_UN;
         case GT_MUL:
-            return FpMul<double, DoubleTraits>(v0, v1);
-        case GT_DIV:
-            return FpDiv<double, DoubleTraits>(v0, v1);
-        case GT_MOD:
-            return FpRem<double, DoubleTraits>(v0, v1);
+            return VNF_MUL_UN;
 
+        case GT_NOP:
+        case GT_COMMA:
+            return VNFunc(oper);
         default:
             unreached();
     }
 }
 
-// Specialize for float for floating operations, that doesn't involve unsigned.
+ValueNumStore::ValueNumStore(Compiler* comp, CompAllocator alloc)
+    : m_pComp(comp)
+    , m_alloc(alloc)
+    ,
+#ifdef DEBUG
+    m_numMapSels(0)
+    ,
+#endif
+    m_nextChunkBase(0)
+    , m_fixedPointMapSels(alloc, 8)
+    , m_checkedBoundVNs(alloc)
+    , m_chunks(alloc, 8)
+    , m_intCnsMap(nullptr)
+    , m_longCnsMap(nullptr)
+    , m_handleMap(nullptr)
+    , m_floatCnsMap(nullptr)
+    , m_doubleCnsMap(nullptr)
+    , m_byrefCnsMap(nullptr)
+    , m_VNFunc0Map(nullptr)
+    , m_VNFunc1Map(nullptr)
+    , m_VNFunc2Map(nullptr)
+    , m_VNFunc3Map(nullptr)
+    , m_VNFunc4Map(nullptr)
+{
```

Comment:
Why not put the DEBUG part at the end -- also I think the commas can be inline...

Reply:
OK, this was part of a cut and paste relocation